### PR TITLE
chore: some changes for Groovy and Kotlin: Groovy default version to 4, support Groovy 5

### PR DESCRIPTION
**Pull request by linux-china**
- Update to Groovy default version to 4.0.14
- Support Groovy 5
- Update to Kotlin default version 1.8.22

### CODE REVIEW

File: src/main/java/dev/jbang/source/sources/GroovySource.java [hunk 1]
```diff
@@ -40,14 +40,15 @@ protected List<String> getNativeOptions() {
 	protected List<String> getRuntimeOptions() {
 		List<String> gopts = Collections.singletonList("-Dgroovy.grape.enable=false");
 		List<String> opts = tagReader.collectOptions("JAVA_OPTIONS", "RUNTIME_OPTIONS");
+		opts.remove("-g");
```
Comment (quintesse) [inline src/main/java/dev/jbang/source/sources/GroovySource.java:43]: This is definitely not the way to handle this :-)

But I'm pretty sure this was already fixed by #1657

Comment (maxandersen) [reply src/main/java/dev/jbang/source/sources/GroovySource.java:43]: yeah - this should not be needed.

**Comment by linux-china** [comment]
> How does user run groovy 3 based scripts with this change ?

//GROOVY 3.0.19

File: src/main/java/dev/jbang/source/sources/GroovySource.java (hunk 1 above)
Comment (linux-china) [reply src/main/java/dev/jbang/source/sources/GroovySource.java]: Done, and delete this line.

**Comment by maxandersen** [comment]
@linux-china sorry missed your comment. could you in this PR then please update the docs feeding https://www.jbang.dev/documentation/guide/latest/usage.html#running-groovy-groovy-experimental and document the option to use //GROOVY for specific versions?

**Comment by maxandersen** [comment]
i've added the docs so we can get this out.

**Pull request merged**